dlt/destinations/impl/redshift/redshift.py

```diff
@@ -65,9 +65,11 @@ def __init__(
         file_path: str,
         staging_credentials: Optional[CredentialsConfiguration] = None,
         staging_iam_role: str = None,
+        staging_region_name: str = None,
```

Comment:
this would have to be made part of the `RedshiftClientConfiguration` and then passed to the load-job

Reply:
2nd thought here: if we add to the configuration, we will have to add a lot of tests, i assume so maybe let me have a chat with my colleagues here
@sh-rp

Reply:
okay so let's remove this line and always take the region from the credentials object

Reply:
Hey @djudjuu && @sh-rp 
Sorry for asking again, but I'm a bit confused. Should we include the region when self._staging_iam_role is used? If not, where should the region value come from?

Reply:
hey no worries,
the code is not making it obvious here, as we only support staging from s3, its safe to assume that the staging credentials will be there. 
so unless they're None (in which case staging is not being used and thus the region doesnt need to be specified) get the region from them and attach them to the credentials, which you can define either with iam_role or access_key&secret.

  try to get region from credentials
  if iam_role -> defined with iam_role ; region
  else -> define with accesskey&secret ; region

Reply:
this CopyFileLoadJob can only be called from staging context:

![image](https://github.com/user-attachments/assets/597eff44-8964-4e51-b208-a6832887833a)
and 
![image](https://github.com/user-attachments/assets/2de840a0-d10f-43b0-879b-141ea731f1e3)

Reply:
> okay so let's remove this line and always take the region from the credentials object

The only thing that confuses me is the mentioned answer, because as far as I understood it now is that I should not remove the staging_region variable from the Load-Job.
I committed my solution and hopefully it is what you expected.